### ISSUE TITLE
gnt-instance migrate does not work on standard xen (3.2.1) in Lenny

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#131]
Originally reported of Google Code with ID 75.

```
<b>What steps will reproduce the problem?</b>
1. gnt-instance migrate instancename.domain.tld

<b>What is the expected output? What do you see instead?</b>

# gnt-instance migrate vinst1.test.org
Instance vinst1.test.org will be migrated. Note that migration is
**experimental** in this version. This might impact the instance if
anything goes wrong. Continue?
y/[n]/?: y
Tue Oct 20 19:29:21 2009 * checking disk consistency between source and target
Tue Oct 20 19:29:22 2009 * switching node gnode2.test.org to secondary mode
Tue Oct 20 19:29:22 2009 * changing into standalone mode
Tue Oct 20 19:29:22 2009 * changing disks into dual-master mode
Tue Oct 20 19:29:23 2009 * wait until resync is done
Tue Oct 20 19:29:23 2009 * preparing gnode2.test.org to accept the instance
Tue Oct 20 19:29:23 2009 * migrating instance to gnode2.test.org
Tue Oct 20 19:29:35 2009 * switching node gnode2.test.org to secondary mode
Tue Oct 20 19:29:35 2009 * changing into standalone mode
Tue Oct 20 19:29:35 2009 * changing disks into single-master mode
Tue Oct 20 19:29:57 2009 * wait until resync is done
Failure: command execution error:
Could not migrate instance vinst1.test.org: Failed to migrate instance:
Failed to migrate instance vinst1.test.org: 
Usage: xm migrate <Domain> <Host>

Migrate a domain to another machine.

Options:

-h, --help           Print this help.
-l, --live           Use live migration.
-p=portnum, --port=portnum
                     Use specified port for migration.
-r=MBIT, --resource=MBIT
                     Set level of resource usage for migration.

Error: can't connect: Connection refused


<b>What version of the product are you using? On what operating system?</b>

Debian Lenny 5.0.3, ganeti 2.0.4, xen 3.2.1, drbd 8.0.14-2

 
<b>Please provide any additional information below.</b>

It was already said: xen 3.2.1 is the standard xen distribution for Debian
Lenny, so it is difficult to get rid of it. Where to start if one want to
help giving ganeti support for xen 3.2.x tools chain?


```
Originally added on 2009-10-20 15:35:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Error: can't connect: Connection refused

Did you configure  /etc/xen/xend-config.sxp to allow migration inside your node
network? In particular you'll need to make sure you have:
(xend-relocation-server yes)
(xend-relocation-port 8002)
(xend-relocation-address '')
(xend-relocation-hosts-allow '^regexp-matching-your-node-ips$')
(for example, if all your nodes have addresses in 192.168.3.0/24 for their secondary ips:
(xend-relocation-hosts-allow '^192\\.168\\.3\\.[0-9]+$')
)

Thanks,

Guido

```
Originally added on 2009-10-20 15:43:55 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thank you! Migration worked just fine!
Sorry for not having checked before the xen conf, but I've never used live migration
on xen and I fooled myself reading the requirements.

Thanks again
davide
```
Originally added on 2009-10-20 16:25:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been fixed in 2.1 with a documentation update.

Thanks,

Guido

```
Originally added on 2009-10-20 17:00:46 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.1

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
